src/connectionController.ts

```diff
@@ -362,6 +356,30 @@ export default class ConnectionController {
     return this._connect(connection.id, connectionType);
   }
 
+  /**
+   * In older versions, we'd manually store a connectionString with an appended
+   * appName with the VSCode extension name and version. This checks for this and
+   * removes the previously appended appName.
+   */
+  private _clearLegacyAppNameOverride(connectionId: string): void {
+    const connectionString = new ConnectionString(
+      this._connections[connectionId].connectionOptions.connectionString
+    );
+
+    if (
+      connectionString.searchParams
+        .get('appname')
+        ?.match(/mongodb-vscode \d\.\d\.\d.*/)
+    ) {
+      connectionString.searchParams.delete('appname');
+      this._connections[connectionId].connectionOptions.connectionString =
+        connectionString.toString();
+      void this._connectionStorage.saveConnection(
```

Comment:
not a fan of this but the async call otherwise messes with the timing of multiple other tests. I'd probably like to change this to be a proper async function instead when I'm back but feel free to take over if needed.